inc/search.php

```diff
@@ -67,6 +70,49 @@ function search(&$data,$base,$func,$opts,$dir='',$lvl=1,$sort='natural'){
     }
 }
 
+
+/**
+ * Checks if a file is a namespace default file according
+ * to https://www.dokuwiki.org/namespaces#namespace_default_linking
+ *
+ * @param  string  $path  full path to the file to be checked
+ * @author Frank Thommen <frank.thommen@gmx.net>
+ */
+function is_ns_index($path='') {
+    if ( !$path or
+         (substr($path,-4) != '.txt') ) {
+        // no path given or not a DW pagefile
+        return false;
+    }
+
+    // /path/to/the/file.txt
+    // +-----------++--+
+    //      ^         ^
+    //      |         $fname
+    //      $basedir
+    // +---------------+
+    //      ^
+    //      $ns
+
+    $ns      = preg_replace('/.txt$/', '', $path);
+    preg_match('|^(.+/)([^/]+)$|', $ns, $matches);
+    $basedir = $matches[1];
+    $fname   = $matches[2];
+
+    if ($fname == $conf['start']) {
```

Comment:
Hm. thanks.  But it still worked (using DokuCMS as testbed) when chaning the default namespace file to an other name (e.g. `afile`)...